client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { curry } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import HeaderCake from 'components/header-cake';
+import { closeAddCardDialog } from 'woocommerce/woocommerce-services/state/label-settings/actions';
+import { getLabelSettingsForm } from 'woocommerce/woocommerce-services/state/label-settings/selectors';
+import CreditCardForm from 'blocks/credit-card-form';
+import { addStoredCard } from 'state/stored-cards/actions';
+import { createCardToken } from 'lib/store-transactions';
+import analytics from 'lib/analytics';
+
+class AddCardDialog extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		isVisible: PropTypes.bool,
+		addStoredCard: PropTypes.func.isRequired,
+		closeAddCardDialog: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.createCardToken = curry( createCardToken )( 'card_add' );
+	}
+
+	recordFormSubmitEvent() {
+		analytics.tracks.recordEvent( 'calypso_add_credit_card_form_submit' );
+	}
+
+	render() {
+		const { siteId, isVisible, translate } = this.props;
+
+		const onClose = () => this.props.closeAddCardDialog( siteId );
+
+		return (
+			<Dialog
+				additionalClassNames="add-credit-card-modal woocommerce wcc-root"
+				isVisible={ isVisible }
+				onClose={ onClose }
+				>
+				<HeaderCake onClick={ onClose }>{ translate( 'Add Credit Card' ) }</HeaderCake>
```

Comment:
This is different than the rest of the modals on Store. They usually have a `Close` button next to the `Save` button. Are OK having this one be different?

Reply:
Ok, I see now that the "Save" button is part of the `CreditCardForm` component. That makes this change harder.

Reply:
This matches the /me/purchases/add-credit-card UI, although I'm not sure if that's much of an advantage. The real advantage from my perspective was being able to use the `<CreditCardForm>` block directly, although a design decision on this would take precedence — we might then just have to copy some of the code out of that block.

Reply:
> The real advantage from my perspective was being able to use the <CreditCardForm> block directly, although a design decision on this would take precedence — we might then just have to copy some of the code out of that block.

I think using the block takes precedence. If design is important maybe we can add an extra prop to the `CreditCardForm` block so it can have an optional `onCancel` callback, and if it's present, a `Cancel` button is rendered.